### PR TITLE
docs: custom container configuration

### PR DESCRIPTION
**Pull request by bartsmykla**
Add custom container configuration docs

Did you sign your commit? [Instructions](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md#sign-your-commits): 👍 

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)?: 👍

### CODE REVIEW

File: docs/docs/dev/explore/dpp-on-kubernetes.md [hunk 1]
```diff
@@ -116,3 +116,147 @@ On Kubernetes, `Dataplane` resource is automatically created by kuma-cp. For eac
 
 When a Pod is deleted its matching `Dataplane` resource is deleted as well. This is possible thanks to the
 [owner reference](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/) set on the `Dataplane` resource.
+
+## Custom Container Configuration
+
+For the vast majority of situations, using the default configuration of injected
+by Kuma containers (`kuma-sidecar` and `kuma-init`) is suggested. If you want to
+modify it the new Kubernetes CRD named `ContainerPatch` was introduced.
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp-on-kubernetes.md:124]:
```suggestion
modify it the Kubernetes CRD named `ContainerPatch` was introduced.
```

File: docs/docs/dev/explore/dpp-on-kubernetes.md [hunk 2]
```diff
@@ -116,3 +116,147 @@ On Kubernetes, `Dataplane` resource is automatically created by kuma-cp. For eac
 
 When a Pod is deleted its matching `Dataplane` resource is deleted as well. This is possible thanks to the
 [owner reference](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/) set on the `Dataplane` resource.
+
+## Custom Container Configuration
+
+For the vast majority of situations, using the default configuration of injected
+by Kuma containers (`kuma-sidecar` and `kuma-init`) is suggested. If you want to
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp-on-kubernetes.md:123]: Maybe move this in a tip.

Something like:

::: warning
In the vast majority of cases you shouldn't need to override the sidecar and init-container configuration. container-patches is a feature which requires good understanding of both Kuma and Kubernetes.
:::

**Review by lahabana — APPROVED** [review]

**Comment by lahabana** [comment]
You are missing adding the new annotation in the k8s annotation reference doc with a link to this new section

File: docs/docs/dev/explore/dpp-on-kubernetes.md (hunk 2 above)
Comment (bartsmykla) [reply docs/docs/dev/explore/dpp-on-kubernetes.md]: addressed

File: docs/docs/dev/explore/dpp-on-kubernetes.md (hunk 1 above)
Comment (bartsmykla) [reply docs/docs/dev/explore/dpp-on-kubernetes.md]: addressed

**Comment by bartsmykla** [comment]
> You are missing adding the new annotation in the k8s annotation reference doc with a link to this new section

added

**Pull request merged**